NeuroID/src/main/java/com/neuroid/tracker/callbacks/NIDCallActivityListener.kt

```diff
@@ -0,0 +1,131 @@
+package com.neuroid.tracker.callbacks
+
+import android.Manifest
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.os.Build
+import android.telephony.PhoneStateListener
+import android.telephony.TelephonyCallback
+import android.telephony.TelephonyManager
+import androidx.annotation.RequiresApi
+import androidx.core.app.ActivityCompat
+import com.neuroid.tracker.events.CALL_IN_PROGRESS
+import com.neuroid.tracker.models.NIDEventModel
+import com.neuroid.tracker.storage.getDataStoreInstance
+import com.neuroid.tracker.utils.NIDLog
+
+class NIDCallActivityListener : BroadcastReceiver() {
+    lateinit var intentFilter: IntentFilter
+    lateinit var intent: Intent
+    private val isReceiverRegistered = false
+
+    internal fun setCallActivityListener(context: Context) {
+        if (ActivityCompat.checkSelfPermission(
+                context, Manifest.permission.READ_PHONE_STATE
+            ) == PackageManager.PERMISSION_GRANTED && !isReceiverRegistered
+        ) {
+            NIDLog.d(msg = "Initializing call activity listener")
+            intentFilter = IntentFilter("android.intent.action.PHONE_STATE")
+            context.registerReceiver(this, intentFilter)
+        } else {
+            NIDLog.d(msg = "Permission to listen to call status not found")
```

Comment:
if we don't have permission I think we should send some sort of event noting that

Reply:
Made the event into an enum to track unauthorized call tracking sessions.